typedspark/_core/dataset.py

```diff
@@ -87,7 +87,7 @@ def _add_schema_metadata(self) -> None:
             self.schema[field.name].metadata = field.metadata
 
     @property
-    def typedspark_schema(self) -> T:
+    def typedspark_schema(self) -> Type[T]:
         """Returns the ``Schema`` of the ``DataSet``."""
         return self._schema_annotations  # type: ignore
```

Comment:
Can we remove the `type: ignore` now too?

Reply:
Nope: 
```/Users/markromme/Documents/kaiko/clones/typedspark/typedspark/_core/dataset.py
  /Users/markromme/Documents/kaiko/clones/typedspark/typedspark/_core/dataset.py:92:16 - error: Expression of type "type[Schema]" cannot be assigned to return type "type[T@DataSet]"
    Type "type[Schema]" cannot be assigned to type "type[T@DataSet]" (reportGeneralTypeIssues)```

Reply:
Ugh... Well, at least it's the right type now!